### PR TITLE
User post

### PR DESCRIPTION
**Pull request by IstiakAhmed-1721365**

### CODE REVIEW

**Comment by CW-AnhNH** [comment]
remove all .DS_Store files in your code

File: db/migrate/20220612170820_add_columns_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class AddColumnsToUsers < ActiveRecord::Migration[7.0]
+  def change
+    #add_column :users, :provider, :string
+    #add_column :users, :uid, :string
```
Comment (CW-AnhNH) [inline db/migrate/20220612170820_add_columns_to_users.rb:4]:
```ruby
    #add_column :users, :provider, :string
    #add_column :users, :uid, :string
    -->
    add_column :users, :provider, :string
    add_column :users, :uid, :string
```

File: db/migrate/20220613045406_add_group_id_to_groups.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddGroupIdToGroups < ActiveRecord::Migration[7.0]
```
[the remark this answers is not in the document]
Comment (IstiakAhmed-1721365) [reply db/migrate/20220613045406_add_group_id_to_groups.rb:1]: should I delete the migration file or just remove that line?

File: app/views/admin/posts/_post.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<div id="<%= dom_id post %>">
```
[the remark this answers is not in the document]
Comment (IstiakAhmed-1721365) [reply app/views/admin/posts/_post.html.erb:1]: If we need to address multiple instances of the same class in the same view, we can prefix the dom_id.

File: app/views/groups/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            <% else %>
+              <%= group.description %>
+            <% end %>
+        </td>
+        <td>
+          <% if group.privacy==0 %>
+              <%= "Private"%>
+            <% else %>
+                <%= "Public" %> 
+              <%end %>
```
[the remark this answers is not in the document]
Comment (IstiakAhmed-1721365) [reply app/views/groups/index.html.erb:53]: what should I change here, Anh-san?

File: app/views/groups/list_for_user.html.erb [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      <th>Created At</th>
+      <th>Total Members</th>
+  
+      <th colspan="2"></th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <% @groups.each do |group| %>
+        <% if group.privacy==1 || current_user.id == group.user_id %>
```
[the remark this answers is not in the document]
Comment (IstiakAhmed-1721365) [reply app/views/groups/list_for_user.html.erb:31]: Users can only see public groups or the private groups that they created, so I checked it with a condition

File: app/controllers/groups_controller.rb [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  before_action :set_group, only: %i[show edit update destroy]
+  before_action :must_be_admin, only: %i[edit update destroy]
+
+
+  # GET /groups or /groups.json
+  def index
+    # @groups = Group.all
+    @q = Group.ransack(params[:q])
+    @groups = @q.result.page(params[:page])
+    @user = User.all
```
[the remark this answers is not in the document]
Comment (IstiakAhmed-1721365) [reply app/controllers/groups_controller.rb:13]: I've removed it

File: app/controllers/admin/comments_controller.rb [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        if @comment.save
+          redirect_to comment_url(@comment), notice: 'Comment was successfully created.'
+        else
+          render :new
+        end
+      end
+    end
+
+    def update
+      respond_to do |_format|
```
Comment (CW-AnhNH) [inline app/controllers/admin/comments_controller.rb:33]: remove `respond_to do |_format|`

File: app/controllers/admin/groups_controller.rb [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+    def new
+      @group = Group.new
+    end
+
+    def edit; end
+
+    def create
+      @group = current_user.groups.build(group_params)
+      respond_to do |_format|
```
Comment (CW-AnhNH) [inline app/controllers/admin/groups_controller.rb:22]: remove `respond_to do |_format|`

File: app/controllers/admin/posts_controller.rb [hunk 9]
```diff
@@ -2,53 +2,56 @@
...
+      @comments = @q.result.includes(:user).page(params[:page])
+    end
+
     def create
-      @post = Post.new(post_params.merge(user_id: current_user.id))
+      @post = Post.new(post_params)
 
       if @post.save
-        redirect_to admin_posts_path
+        redirect_to post_url(@post), notice: 'Post was successfully created.'
```
Comment (CW-AnhNH) [inline app/controllers/admin/posts_controller.rb:27]: What is the difference between path and url? why do you change the path to the url?

File: app/views/admin/posts/_form.html.erb [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+  <% end %>
+
+</div>
+
+
+<b>Group: </b></br>
+<%= collection_select(:post, :user_id, Group.all, :id, :name) %>
+
+
+
```
Comment (CW-AnhNH) [inline app/views/admin/posts/_form.html.erb:66]: Please check and remove empty lines of all files, like this

File: db/migrate/20220613045406_add_group_id_to_groups.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (CW-AnhNH) [reply db/migrate/20220613045406_add_group_id_to_groups.rb:1]: you should delete this file

File: db/seeds.rb [hunk 11]
```diff
@@ -5,12 +5,18 @@
...
 
 (1..10).each do |id|
     username = Faker::Esport.player
     family_name = Faker::Name.unique.name
     last_name = Faker::Name.unique.name
     birthday = Faker::Date.birthday(min_age: 18, max_age: 65)
+>>>>>>> c37e6b454a3cb9e23eb3f75ce1504dfe1237f6fb
     email = Faker::Internet.unique.email
     comment = Faker::Lorem.sentence(word_count: 3)
     post_title = Faker::Coffee.blend_name
```
Comment (CW-AnhNH) [inline db/seeds.rb:22]: you have some errors here when using git rebase
```ruby
<<<<<<< HEAD
(1..10).each do |id|
    username = Faker::Esport.player
    last_name = Faker::Name.unique.name
=======

(1..10).each do |id|
    username = Faker::Esport.player
    family_name = Faker::Name.unique.name
    last_name = Faker::Name.unique.name
    birthday = Faker::Date.birthday(min_age: 18, max_age: 65)
>>>>>>> c37e6b454a3cb9e23eb3f75ce1504dfe1237f6fb

---->

(1..10).each do |id|
    username = Faker::Esport.player
    family_name = Faker::Name.unique.name
    last_name = Faker::Name.unique.name
    birthday = Faker::Date.birthday(min_age: 18, max_age: 65)

```

**Review by CW-HauNC — APPROVED** [review]

**Pull request merged**